sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ValuePostprocessor.scala

```diff
@@ -218,4 +220,68 @@ final class ValueEnricher(
     enrichTransaction(incompleteTx.transaction).map(transaction =>
       incompleteTx.copy(transaction = transaction)
     )
+
+  /*
+    This will renormalize a contract w.r.t a (up/down)grade template.
+    In practice this potentially
+     - adds/removes none fields
+     - change the transaction version.
+   */
+  private[this] def reversionValue(
+      typ: Ast.Type,
+      value: Value,
+      dstVer: TransactionVersion,
+  ): Result[Value] =
+    translateValue(typ, value).map(_.toNormalizedValue(dstVer))
+
+  // This
+  def reversion(
+      contract: FatContractInstance,
+      dstTmplId: Ref.TypeConName,
+  ): Result[FatContractInstance] = {
+    if (contract.templateId == dstTmplId)
```

Comment:
Would not work for pre-converted contracts.

Reply:
@remyhaemmerle-da please double-check there is no problem with the engine. That said, up to you how you handle things.